homeassistant/components/totalconnect/binary_sensor.py

```diff
@@ -77,16 +130,129 @@ def device_class(self):
             return BinarySensorDeviceClass.MOTION
         if self._zone.is_type_medical():
             return BinarySensorDeviceClass.SAFETY
-        return None
+        # "security" type is a generic category so test for it last
+        if self._zone.is_type_security():
+            return BinarySensorDeviceClass.DOOR
+
+        raise HomeAssistantError(
+            f"TotalConnect zone {self._zone_id} reported an unexpected device class."
+        )
+
+
+class TotalConnectLowBatteryBinarySensor(TotalConnectBinarySensor):
+    """Represent an TotalConnect zone low battery status."""
+
+    def __init__(self, zone_id, location_id, zone):
+        """Initialize the TotalConnect status."""
+        self._zone_id = zone_id
+        self._location_id = location_id
+        self._zone = zone
+        self._name = f"{self._zone.description} low battery"
+        self._unique_id = f"{location_id} {zone_id} low battery"
```

Comment:
Store the name and key inside the entity description and you can make a generic binary sensor entity class that can handle all entity descriptions.

Reply:
thanks for the suggestion, it's much cleaner now